### PR TITLE
plugin-samples: Capital plugin

### PR DESCRIPTION
**Pull request by jackiechui**
Adding a simple sample plugin to demonstrate how to make network requests to populate parameter suggestions in query mode.
The plugin shows the capital city of the selected country in a visual bell when run.

image

Test Plan:

- Import and run the plugin
- Review code and comments

### CODE REVIEW

File: capital/code.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        name: country.name,
+        data: {
+          index: index,
+          name: country.name,
+          capital: country.capital,
+        },
+      };
+    });
+    result.setSuggestions(countries);
+  };
```
Comment (rohitchouhan8) [inline capital/code.ts:34]: Pretty minor, but I don't think this needs to be in the figma.parameters.on() method. Perhaps move it outside just to show that they aren't actually related?

Comment (jackiechui) [reply capital/code.ts:34]: I put it in the method to have access to `result`. Can you show me how I can take it out and call `result.setSuggestions`?

Comment (jackiechui) [reply capital/code.ts]: Good idea! I've moved it out.

File: capital/code.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      name: country.name,
+      data: {
+        index: index,
+        name: country.name,
+        capital: country.capital,
+      },
+    };
+  });
+  // Set the suggestions after the a message has been received from the iframe
+  resultObj.setSuggestions(countries);
```
Comment (johndoherty) [inline capital/code.ts:29]: I think there are a couple possible race conditions here:
- If you type a character before the countries are loaded, the countries will never be loaded for that input event
- If the countries are loaded before the first input event, then they will never be loaded for that input event.

I think the safer way of doing this is to make `countries` a promise that can be awaited in the `input` event handler. Something like:

```ts
let resolveCountries = (countries: string[]) => {}
const countriesPromise = new Promise<string[]>((resolve) => {resolveCountries = resolve})
// ...
figma.ui.onmessage = (json) => {
  // ...
 resolveCountries(countries)
}
// on('input'):
result.setLoadingMessage('Loading countries...');
const countries = await countriesPromise
result.setSuggestions(/* filter countries */)
```

Comment (jackiechui) [reply capital/code.ts]: Thanks! I'll update as you've suggested.

**Review by johndoherty — APPROVED** [review]
Nice! LGTM

**Pull request merged**